notebooks/PFT-Gridding.ipynb

```diff
@@ -0,0 +1,401 @@
+{
```

Comment:
<p>I think you can skip the for loop here and take advantage of numpy's vectorized indexing:</p><p> </p><pre class="ql-syntax" spellcheck="false">gridded[:, vegtype.values.astype(int) - 1, jxy.values.astype(int) - 1, ixy.values.astype(int)] = gpp.values
</pre><p>This will likely be much, much, much faster! And it all fits in one line.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NCAR/ctsm_py/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Reply:
<p>Thanks for the suggestion! I worked through a slightly modified version of your code:</p><p> </p><p>gridded[:, vegtype.values.astype(int), jxy.values.astype(int) - 1, ixy.values.astype(int) - 1] = gpp.values</p><p> </p><p>One related comment/question: simplifying to this one line solution removes the loop over the cellwt_float variable, which essentially masks out the cells without any area for that PFT. It still produces a file that looks (to my eye) very similar to the loop. I can't seem to difference the files created each way (requires too much memory) to verify that there are no differences, though. Do you think that this will cause any noticeable differences?</p>

Reply:
<p>My understanding is that these two methods should produce identical (bit-for-bit) results. If you are concerned about the cell weights, you could add a subset the 1d vectors before setting the values to the full array:</p><p> </p><pre class="ql-syntax" spellcheck="false">mask1d = cellwt_float &gt; 0
i = ixy.values.astype(int)[mask1d] - 1
j = ixy.values.astype(int)[mask1d] - 1
v = vegtype.values.astype(int)[mask1d] - 1

gridded[:, v, j, i] = gpp.values[mask1d]
</pre>